File: day2/part2.py
def check(line, skip):
    i = 0
    lastnum = 0
    valid = True
    isDecreasing = True
    arr = [int(x) for x in line.split(" ")]
    if(skip != -1):
        arr.pop(skip)
    for num in arr:
        print(f"{num}, ", end="")
        if (i == 0):
            lastnum = num
        else:
            if (abs(num - lastnum) > 3 or num == lastnum):
                valid = False
                break
            if (i == 1):
                isDecreasing = num < lastnum
            else:
                if isDecreasing and num > lastnum:
                    valid = False
                    break
                if (not isDecreasing) and num < lastnum:
                    valid = False
                    break
        lastnum = num
        i += 1
    return valid


f = open("input.txt", "r")
res = 0
for line in f.readlines():
    valid = False
    for i in range(len(line.split(" "))):
        print(i)
        valid = check(line, i)
        if (valid == True):
            break
    if valid == False:
        valid = check(line, -1)
    if valid:
        res += 1
            
        
    print("")
print(f"res = {res}")

File: day2/part2.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <stdbool.h>

const char* filename = "input.txt";

const int numLines = 1000;

int checkVals(char *curLine, char skip) {
    size_t i = 0;
    printf("\nskip == %d ", skip);
    char *lineCopy = malloc(strlen(curLine) + 1);
    strcpy(lineCopy, curLine);
    char *subString = strtok(lineCopy, " ");
    if (skip == 0) {
        subString = strtok(NULL, " ");
    }
    int lastVal = 0;
    bool isDecreasing = false;
    while (subString != NULL) {

        if (i == 0) {
            lastVal = atoi(subString);
            printf("%d, ", lastVal);
        } else {
            int curVal = atoi(subString);
            printf("%d, ", curVal);
            if(abs(curVal - lastVal) > 3 || curVal == lastVal) {
                return 0;
            }
            if (i == 1) {
                if (lastVal > curVal) {
                    isDecreasing = true; 
                }
            } else {
                if (isDecreasing && lastVal < curVal) {
                    return 0;
                }
                if (!isDecreasing && lastVal > curVal) {
                    return 0;
                }
            }
            lastVal = curVal;
        }
        subString = strtok(NULL, " ");
        i++;
        if (i == skip) {
            subString = strtok(NULL, " ");
        }
    }
    free(lineCopy);
    return 1;
}

int main() {
    FILE *fptr;

    // Open a file in read mode
    fptr = fopen(filename, "r");

    if(fptr == NULL) {
        printf("\tfile %s not found\n", filename);
    }

    // Store the content of the file
    char curLine[256];

    int pos = 0;

    int res = 0;
    // Read the content and print it
    while(fgets(curLine, 256, fptr)) {
        for (int i = 0; i < 11; i++) {
            if(checkVals(curLine, i) == 1) {
                res++;
                break;
            }
        }
        pos++;
    }

    printf("result = %d\n", res);


    // Close the file
    fclose(fptr);
}

